core/src/main/java/org/elasticsearch/common/bytes/BytesReference.java

```diff
@@ -170,4 +88,54 @@ public BytesRef next() throws IOException {
         };
     }
 
+    static boolean bytesEqual(BytesReference a, BytesReference b) {
+        if (a == b) {
+            return true;
+        }
+        if (a.length() != b.length()) {
+            return false;
+        }
+
+        return bytesEquals(a, b);
+    }
+
+    static boolean bytesEquals(BytesReference a, BytesReference b) {
```

Comment:
This is confusing to have these two methods with the same params and name differing based on plural...can we at least have some javadocs if both are really needed?